core/controllers/reader.py

```diff
@@ -119,29 +119,26 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
-def _get_exploration_player_data(self, exploration_id, version, collection_id):
+def _get_exploration_player_data(
+    exploration_id, version, collection_id, can_edit):
 
     try:
         exploration = exp_services.get_exploration_by_id(
             exploration_id, version=version)
-    except Exception as e:
-        raise self.PageNotFoundException(e)
+    except Exception:
```

Comment:
You don't really need the try/except here. However, after removing it, can you add a comment above the "exploration = ..." line that says "If the exploration does not exist, an Exception is raised."